### PR TITLE
Add support for Python 3.11 and deprecate Python 3.8

### PR DESCRIPTION
**Pull request by karan6181**
## Description of changes:
- Add support for Python 3.11
- Deprecate Python 3.8 support since EOL is 10/2024 and the streaming dataset supports up to 3 Python minor versions.
- Add linting run for Python 3.10 and Python 3.11.

## Testing
- CORGI Integration test passes: `streaming-integration-tests-runner-L3YsGu`
- CORGI Resnet `resnet-imagenet1k-1-node-convergence-5LRs3w` and SD2 `sd2-base-256-1-node-convergence-lKoPUa` test passes

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [ ] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#submitting-a-contribution))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: simulation/interfaces/widgets.py [hunk 1]
```diff
@@ -58,18 +58,19 @@ def get_line_chart(data: pd.DataFrame,
...
+            y='throughput (batches/s)' if throughput else 'cumulative network usage (bytes)',
+            opacity=alt.condition(hover, alt.value(0.3), alt.value(0)),  # pyright: ignore
+            tooltip=[
+                alt.Tooltip('step', title='Step'),
+                alt.Tooltip(
+                    'throughput (batches/s)' if throughput else 'cumulative network usage (bytes)',
+                    title='Throughput' if throughput else 'Network Usage'),
+            ],
+        ).add_params(hover))
+    return (lines + points + tooltips).interactive()  # pyright: ignore
```
Comment (knighton) [inline simulation/interfaces/widgets.py:73]: this 3.11 lint warning has been driving me crazy for a long time... brutal, but effective

**Review by knighton — APPROVED** [review]

File: simulation/interfaces/widgets.py (hunk 1 above)
Comment (karan6181) [reply simulation/interfaces/widgets.py:73]: @knighton How did you solve it? Maybe I can just copy-paste here?

Comment (snarayan21) [reply simulation/interfaces/widgets.py:73]: It's a pretty hairy linting error right...i think we should just leave the `# pyright: ignore`

**Review by snarayan21 — APPROVED** [review]
lgtm

**Pull request merged**